error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/FileRules.java

```diff
@@ -174,4 +177,70 @@ boolean after(File file) {
       return file.mkdirs() || file.exists();
     }
   }
+
+  /**
+   * Prefer {@link Files#newBufferedReader(Path)} over creating a {@link BufferedReader} from a
+   * {@link FileInputStream} for a string path.
+   */
+  static final class FilesNewBufferedReaderFromStringPath {
+    @BeforeTemplate
+    @SuppressWarnings("DefaultCharset" /* This violation will be rewritten. */)
+    BufferedReader before(String path) throws IOException {
+      return new BufferedReader(new InputStreamReader(new FileInputStream(path)));
+    }
+
+    @AfterTemplate
+    BufferedReader after(String path) throws IOException {
+      return Files.newBufferedReader(Path.of(path));
```

Comment:
We can actually add one more case here: dropping `UTF-8` when specified explicitly. (This one may be slightly controversial, but in this repo we _generally_ prefer dropping defaults when possible.)